### PR TITLE
Added a way to accept join requests

### PR DESCRIPTION
**Pull request by Mart1nux**
Added a page that has a list of all the people that want to join this platform with their personal information and an option to accept their request or to decline it.

Page:
image

### CODE REVIEW

File: Pages/PendingVolunteersList.razor.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            PendingVolunteer pendingVolunteer = m_karmaContext.PendingVolunteers.Where(x => x.Id == id).FirstOrDefault();
+            var volunteer = new Volunteer(pendingVolunteer.Name, pendingVolunteer.Surname, Guid.NewGuid());
+            int result = m_DBServiceProvider.AddToDB(volunteer);
+            if (result == 0)
+                m_notifactionTransmitter.ShowMessage("Volunteer added", MatToastType.Success);
+            else if (result == -1)
+                m_notifactionTransmitter.ShowMessage("An error occured while adding volunteer to the database", MatToastType.Danger);
+            m_karmaContext.PendingVolunteers.Where(x => x.Id == id).DeleteFromQuery();
+            m_karmaContext.SaveChanges();
+        }
```
Comment (cezaris13) [inline Pages/PendingVolunteersList.razor.cs:45]: missing \n between methods

File: Pages/PendingVolunteersList.razor.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public partial class PendingVolunteersList
+    {
+        [Inject]
+        public IDBServiceProvider m_DBServiceProvider { get; set; }
+
+        public string FilterValue = "";
+
+        private KarmaContext m_karmaContext = new();
+
+        string CurrentUserId { get; set; }
```
Comment (cezaris13) [inline Pages/PendingVolunteersList.razor.cs:21]: this should have private if it is used in cs file and Public if it is used in razor file(I think). Also is this property even used here?

File: Pages/PendingVolunteersList.razor.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using Microsoft.AspNetCore.Components;
+
+namespace Karma.Pages
+{
+    public partial class PendingVolunteersList
+    {
+        [Inject]
+        public IDBServiceProvider m_DBServiceProvider { get; set; }
+
+        public string FilterValue = "";
```
Comment (cezaris13) [inline Pages/PendingVolunteersList.razor.cs:17]: If it is not a property It should start with lowercase

File: Pages/VolunteerRegistration.razor.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{
+    public partial class VolunteerRegistration
+    {
+        private String Name { get; set; } = "";
+
+        private String Surname { get; set; } = "";
+
+        private String PhoneNumber { get; set; } = "";
+
+        KarmaContext KarmaContext = new KarmaContext();
```
Comment (cezaris13) [inline Pages/VolunteerRegistration.razor.cs:15]: private KarmaContext  m_karmaContext = new();

File: Pages/VolunteerRegistration.razor.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            m_notifactionTransmitter.ShowMessage("Registration Succesful", MatToastType.Success);
+            Name = "";
+            Surname = "";
+            PhoneNumber = "";
+            NavigateToMain();
+        }
+
+        public void NavigateToMain()
+        {
+            m_uriHelper.NavigateTo("/");
```
Comment (cezaris13) [inline Pages/VolunteerRegistration.razor.cs:31]: I see that this is used once in CreatePendingVolunteer(). Could you move m_uriHelper... to this method?(I completely missed this while reviewing previous PR)

File: Pages/VolunteerRegistration.razor.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
+using System;
+using Karma.Models;
+using MatBlazor;
+
+namespace Karma.Pages
+{
+    public partial class VolunteerRegistration
+    {
+        private String Name { get; set; } = "";
```
Comment (cezaris13) [inline Pages/VolunteerRegistration.razor.cs:9]: string

File: Pages/PendingVolunteersList.razor.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            return m_karmaContext.PendingVolunteers;
+        }
+
+        public void AddANewVolunteer(Guid id)
+        {
+            PendingVolunteer pendingVolunteer = m_karmaContext.PendingVolunteers.Where(x => x.Id == id).FirstOrDefault();
+            var volunteer = new Volunteer(pendingVolunteer.Name, pendingVolunteer.Surname, Guid.NewGuid());
+            int result = m_DBServiceProvider.AddToDB(volunteer);
+            if (result == 0)
+                m_notifactionTransmitter.ShowMessage("Volunteer added", MatToastType.Success);
```
[the remark this answers is not in the document]
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor.cs:40]: Yep, this is my fault, fixed it.

File: Pages/PendingVolunteersList.razor [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+            <th>Phone number</th>
+            <th>Actions</th>
+        </tr>
+        @foreach (var potentialVolunteer in GetPendingVolunteers())
+        {
+        var nameSurname = potentialVolunteer.Name + " " + potentialVolunteer.Surname;
+        var surnameName = potentialVolunteer.Surname + " " + potentialVolunteer.Name;
+        var regex = new Regex($"(.*){FilterValue}(.*)", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        var matches = regex.Matches(nameSurname + surnameName);
+        @if (matches.Count > 0)
```
[the remark this answers is not in the document]
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor]: If you mean that 31 line should be a it to the right then this is how ctrl+k, ctrl+d aligned it.

File: Pages/PendingVolunteersList.razor.cs (hunk 1 above)
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor.cs]: Added a new line between methods.

File: Pages/PendingVolunteersList.razor.cs (hunk 2 above)
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor.cs]: Added private. And yes, after changes the property is used.

File: Pages/PendingVolunteersList.razor.cs (hunk 3 above)
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor.cs]: It starts from a lower case now.

File: Pages/VolunteerRegistration.razor.cs (hunk 4 above)
Comment (Mart1nux) [reply Pages/VolunteerRegistration.razor.cs:15]: Done like written.

File: Pages/VolunteerRegistration.razor.cs (hunk 5 above)
Comment (Mart1nux) [reply Pages/VolunteerRegistration.razor.cs:31]: Moved the navigation to the CreatePendingVolunteer().

File: Pages/VolunteerRegistration.razor.cs (hunk 6 above)
Comment (Mart1nux) [reply Pages/VolunteerRegistration.razor.cs:9]: Yeah, I also noticed the capital strings. Changed them.

File: Pages/PendingVolunteersList.razor (hunk 8 above)
[the remark this answers is not in the document]
Comment (cezaris13) [reply Pages/PendingVolunteersList.razor]: Strange, my formatter says otherwise : \

**Review by ArentasMeinorius — APPROVED** [review]

**Comment by ArentasMeinorius** [comment]
next time (if it's possible) solve conflicts before review

**Review by cezaris13 — APPROVED** [review]

File: Pages/PendingVolunteersList.razor (hunk 8 above)
[the remark this answers is not in the document]
Comment (Mart1nux) [reply Pages/PendingVolunteersList.razor]: fixed as your formatter says

**Pull request merged**